src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/CacheHealthDecider.java

```
@@ -16,146 +16,188 @@
+      final ResourceEnum cacheType,
+      final boolean increase) {
+    if (ModifyCacheMaxSizeAction.NAME.equals(actionName)) {
+      return configureCacheMaxSize(
+          esNode, cacheType, increase);
     }
-
-    private ModifyCacheCapacityAction configureCacheCapacity(
-            final NodeKey esNode,
-            final ResourceEnum cacheType,
-            final long currentCapacityInBytes,
-            final boolean increase) {
-        final ModifyCacheCapacityAction action =
-                new ModifyCacheCapacityAction(esNode, cacheType, currentCapacityInBytes, increase);
-        if (action.isActionable()) {
-            return action;
-        }
-        return null;
+    return null;
+  }
+
+  private ModifyCacheMaxSizeAction configureCacheMaxSize(
+      final NodeKey esNode,
+      final ResourceEnum cacheType,
+      final boolean increase) {
+    final double cacheUpperBound = getCacheUpperBound(cacheType);
+    if (cacheUpperBound != -1) {
+      final ModifyCacheMaxSizeAction action =
+          new ModifyCacheMaxSizeAction(
+              esNode,
+              cacheType,
+              getAppContext().getNodeConfigCache(),
+              cacheUpperBound,
+              increase);
+      if (action.isActionable()) {
+        return action;
+      }
     }
-
-    private long getNodeCacheCapacityInBytes(final NodeKey esNode, final ResourceEnum cacheType) {
-        // TODO: use NodeConfigurationRca to return capacity, for now returning random value in Bytes
-        if (cacheType.equals(ResourceEnum.FIELD_DATA_CACHE)) {
-            return 1000L;
-        }
-        return 1000L;
+    return null;
+  }
+
+  private double getCacheUpperBound(final ResourceEnum cacheType) {
+    if (cacheType.equals(ResourceEnum.FIELD_DATA_CACHE)) {
+      return getFieldDataCacheUpperBound();
+    } else if (cacheType.equals(ResourceEnum.SHARD_REQUEST_CACHE)) {
+      return getShardRequestCacheUpperBound();
     }
+    return -1;
```

When will we hit this? We should throw exceptions instead of overloading return values. How about an `IllegalArgumentException` ?

---

Throwing IllegalArgumentException